### ISSUE TITLE
[JAVA] Jackson innerEnum missing JsonValue symbol

### ISSUE THREAD

**Issue by konphorm** [swagger-api/swagger-codegen#8724]
##### Description

When using ```io.swagger:swagger-codegen-cli:2.3.1``` from maven, get build issues for models with an inner enum. Specifically:

```
error: cannot find symbol
@JsonValue
   ^
symbol:   class JsonValue
```

Looking at the mustache file you can see the import is only in the modelEnum, yet the annotation is used in the modelInnerEnum template.

```
virt % grep -R  JsonValue ./Java
./Java/modelInnerEnum.mustache:    @JsonValue
./Java/modelEnum.mustache:import com.fasterxml.jackson.annotation.JsonValue;
./Java/modelEnum.mustache:  @JsonValue
```

The modelInnerEnum.mustache is expanded by the pojo.mustache template, both of which get their includes from model.mustache, which is also the top level for modelEnum.mustache.

##### Swagger-codegen version

From maven: ```io.swagger:swagger-codegen-cli:2.3.1```

This seems to be a regression from ```io.swagger:swagger-codegen-cli:2.2.2``` due to the added annotation.

##### Swagger declaration file content or url

JSON Model:

```
"FooObject" : {
      "type" : "object",
      "properties" : {
        "stringField" : {
          "type" : "string"
        },
        "enumField" : {
          "type" : "string",
          "enum" : [ "One", "Two", "Three", "Four", "Five" ]
        }
      }
    }
```

##### Command line used for generation

generate with a language using Java with jackson.

##### Steps to reproduce

- Model with inner enum
- Generate Java model with Jackson
- Compile

##### Related issues/PRs

https://github.com/swagger-api/swagger-codegen/issues/3611
https://github.com/swagger-api/swagger-codegen/issues/6587

##### Suggest a fix/enhancement

The imports from modelEnum should be moved into model to avoid the issue and be consistent with not having imports in pojo.mustache. Short of that add the imports to pojo.mustache.

So, basically integrate this: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/Java/modelEnum.mustache#L1-L11

into here: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/Java/model.mustache#L18

**Comment by konphorm** [comment]
Looks like someone else has already hit this issue.

https://github.com/swagger-api/swagger-codegen/issues/8719

**Comment by konphorm** [comment]
Actually looks like all of these are already patched in a branch.

https://github.com/swagger-api/swagger-codegen/commit/65ee4cb835049fa0d130e4059041a87d894d69bb#diff-cc3e8eb9a2b26248932102a87608fc60

From this issue: https://github.com/swagger-api/swagger-codegen/pull/2806

**Comment by yeralin** [comment]
@konphorm if it was already patched, why do we still face the issue?

**Comment by philip-rx** [comment]
Is there any practical workaround for this ?

**Comment by uderani29** [comment]
@philip-rx This works when you update the codegen version. Example what worked for me was:

```
	<groupId>io.swagger</groupId>
	<artifactId>swagger-codegen-maven-plugin</artifactId>
	<version>2.4.1</version>
	<configuration>
		<configOptions>
		<library>jersey1</library>
				 <dateLibrary>java8</dateLibrary>
   		<java8>true</java8>
		</configOptions>
	</configuration>

```
Also might have to change 

```
			<groupId>com.fasterxml.jackson.datatype</groupId>
			<artifactId>jackson-datatype-jsr310</artifactId>
			<version>${jackson-version}</version>
```
and remove <artifactId>jackson-datatype-joda</artifactId>
